paasta_tools/cli/cmds/spark_run.py

```diff
@@ -480,10 +490,46 @@ def get_spark_conf_str(
         )
         sys.exit(1)
 
-    spark_conf = list()
-    for opt, val in dict(non_user_args, **user_args).items():
-        spark_conf.append(f'--conf {opt}={val}')
-    return ' '.join(spark_conf)
+    return dict(non_user_args, **user_args)
+
+
+def create_spark_config_str(spark_config_dict):
+    spark_config_entries = list()
+    for opt, val in spark_config_dict.items():
+        spark_config_entries.append(f'--conf {opt}={val}')
+    return ' '.join(spark_config_entries)
+
+
+def emit_resource_requirements(spark_config_dict, paasta_cluster):
+    num_executors = int(spark_config_dict['spark.cores.max']) / int(spark_config_dict['spark.executor.cores'])
+    memory_per_executor = spark_memory_to_megabytes(spark_config_dict['spark.executor.memory'])
+
+    desired_resources = {
+        'cpus': int(spark_config_dict['spark.cores.max']),
+        'mem': memory_per_executor * num_executors,
+        'disk': memory_per_executor * num_executors,  # rough guess since spark does not collect this information
```

Comment:
what if you just set this to None instead of making a guess?  I worry that this might make bin-packing harder by adding more constraints, although maybe that's a good thing?  Also maybe it doesn't matter since we're not really disk-constrained... but on the other hand most of our services underestimate the amount of disk space they use... but on the gripping hand, we can't put hard limits on the amount of disk a service/job uses anyways, so....... *shrug*

Your call what to do here.

Reply:
Yeah I'm not super sure here.  To add a fourth hand (I'm goro?), in some cases that I don't fully understand, Spark will spill a bunch of data to disk, and I _think_ we've seen some jobs eat all the disk on a box before.  Anyway, it seems like it's nice to annotate that these jobs can use some disk, so I guess I'm going to leave it for now?